portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,11 +24,18 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-    // The welcome message
-    response.getWriter().println("<h1>Hello, Jiaxi Chen!</h1>");
-  }
+    private int count = 0;
```

Comment:
1. I would suggest to move private member methods/variables at the end of the class.
```
public class Example {
  public void doGet(...) {}

  private int count = 0;
}
```

2. class member variable is not threadsafe. However, Java servlet will call doGet method concurrently (when the QPS is large), which cause the result unpredictable. Please ensure thread-safety though I know the QPS won't be large in our situation. Java provides several mechanisms for thread-safety (synchronized, lock, ...).
You could consult virtual tech support for details since they are Java masters. I expect they could give better suggestions.

3. Java Servlet could instantiate multiple servlet instances under the production environment (distributed environment), which means there might be multiple ```count``` each with different value. Please refer to [this question](https://stackoverflow.com/questions/22544152/how-many-instances-are-created-for-a-http-servlet) for details.

Reply:
I added the synchronized keyword around the increment and print statement. The keyword will ensure only one thread will access the data at each time, like a lock. I guess it will fix the race condition.

Reply:
Still not solve #3, but we could leave this to the future. It's not a critical issue in the development environment.